lib/msal-node/test/client/ClientCredentialClient.spec.ts

```diff
@@ -982,6 +982,88 @@ describe("ClientCredentialClient unit tests", () => {
         expect(authResult.state).toHaveLength(0);
     });
 
+    it("re-acquires a token via a network request, because the cached token's refreshOn value is expired", async () => {
+        sinon
+            .stub(Authority.prototype, <any>"getEndpointMetadataFromNetwork")
+            .resolves(DEFAULT_OPENID_CONFIG_RESPONSE.body);
+        sinon
+            .stub(
```

Comment:
In terms of tests, you can extend the coverage if instead of mocking some method `executePostToTokenEndpoint` (which btw will break if on rename), to injecting custom HTTP responses or mocking at the HTTP Client level.

Otherwise, please point to tests that show that `executePostToTokenEndpoint` behave like your mock states when getting 500 errors from STS.

Reply:
We never had `http` level mocking. I remember we scheduled it, never got around it.